config/routes.rb

```diff
@@ -1,13 +1,32 @@
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users,
+             controllers: {
+                 sessions: 'users/sessions',
+                 registrations: 'users/registrations'
+             }
 
   root 'users#index'
+  # resources :posts do
+  #   resources :comments
+  # end
+
+  namespace :api do
+    namespace :v1 do
+      resources :posts
+      get 'posts/:post_id/comments/' , to: 'posts#comments_for_post'
+      post 'posts/:post_id/comments/', to: 'posts#create_comment'
+    end
+  end
+
   get '/users', to: 'users#index'
   get '/users/:id', to: 'users#show', as: 'user'
 
+  delete '/users/:user_id/posts/:id', to: 'posts#destroy', as: 'delete_user_post'
+
   get '/users/:user_id/posts/new', to: 'posts#new', as: 'new_user_post'
   post '/users/:user_id/posts', to: 'posts#create', as: 'user_posts_create'
 
+  delete '/users/:user_id/posts/:post_id/comments/:comment_id', to: 'comments#destroy', as: 'delete_user_post_comment'
   get '/users/:user_id/posts/:post_id/comments/new', to: 'comments#new', as: 'new_user_post_comment'
   post '/users/:user_id/posts/:post_id/comments', to: 'comments#create', as: 'user_post_comments'
```

Comment:
[OPTIONAL] You could have used nested resources to achieve this. Kindly check this[ link](https://www.digitalocean.com/community/tutorials/how-to-create-nested-resources-for-a-ruby-on-rails-application) for a better understanding of nested resources